test/jdk/javax/swing/JPasswordField/PasswordFieldInputMapWordTest.java

```diff
@@ -0,0 +1,107 @@
+ * questions.
+ */
+
+/**
+ * @test
+ * @key headful
+ * @bug 8358813
+ * @summary Password fields' InputMap should not include any word-related action.
+ *
+ * @run main PasswordFieldInputMapWordTest
+ */
+
+import javax.swing.InputMap;
+import javax.swing.JComponent;
+import javax.swing.JPasswordField;
+import javax.swing.KeyStroke;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import javax.swing.UnsupportedLookAndFeelException;
+import javax.swing.text.DefaultEditorKit;
+
+public class PasswordFieldInputMapWordTest {
+    public static void main(String[] args) throws Exception {
+        for (UIManager.LookAndFeelInfo laf :
+                UIManager.getInstalledLookAndFeels()) {
+            System.out.println("Testing LAF: " + laf.getClassName());
+            SwingUtilities.invokeAndWait(() -> {
+                if (setLookAndFeel(laf)) {
+                    runTest();
+                }
+            });
+        }
+    }
+
+    private static boolean setLookAndFeel(UIManager.LookAndFeelInfo laf) {
+        try {
+            UIManager.setLookAndFeel(laf.getClassName());
+            return true;
+        } catch (UnsupportedLookAndFeelException e) {
+            System.err.println("Skipping unsupported look and feel:");
+            e.printStackTrace();
+            return false;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static void runTest() {
+        JPasswordField field = new JPasswordField();
+
+        boolean testPassed = true;
+        for (int condition : new int[] {
+                JComponent.WHEN_IN_FOCUSED_WINDOW,
+                JComponent.WHEN_FOCUSED,
+                JComponent.WHEN_ANCESTOR_OF_FOCUSED_COMPONENT
+        }) {
+            InputMap inputMap = field.getInputMap(condition);
+            if (inputMap.allKeys() == null) {
+                continue;
+            }
+            for (KeyStroke keyStroke : inputMap.allKeys()) {
+                Object actionBinding = inputMap.get(keyStroke);
+                // these are all the actions with "word" in the name:
+                if (actionBinding == DefaultEditorKit.deleteNextWordAction ||
+                        actionBinding == DefaultEditorKit.deletePrevWordAction ||
+                        actionBinding == DefaultEditorKit.beginWordAction ||
+                        actionBinding == DefaultEditorKit.endWordAction ||
+                        actionBinding == DefaultEditorKit.selectionBeginWordAction ||
+                        actionBinding == DefaultEditorKit.selectionEndWordAction ||
+                        actionBinding == DefaultEditorKit.previousWordAction ||
+                        actionBinding == DefaultEditorKit.nextWordAction ||
+                        actionBinding == DefaultEditorKit.selectionPreviousWordAction ||
+                        actionBinding == DefaultEditorKit.selectionNextWordAction ) {
```

Comment:
Does it make sense to put the actions in a list or array? Then `listOfActions.contains(actionBinding)` instead of this long `if` condition?

Reply:
I don't have a preference. I pushed an update that wraps them in a TreeSet.